### PR TITLE
Added plugins, makefile, and travis

### PR DESCRIPTION
**Pull request by mcarmonaa**
- This PR add some useful `sbt plugins` (dependency-graph, scalastyle, checkstyle, coverage and eclipse) and a `Makefile` as well.
- It modifies travis.yml to make use of the plugins too.
- Added sbt script wrapper

This changes address some of the tasks pointed in https://github.com/src-d/siva-java/issues/8

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+
+# Rules
+all: clean build
+
+.PHONY: fatjar test clean
+build:
+	$(SBT) package
+
+fatjar:
```
Comment (erizocosmico) [inline Makefile]: why do we need a fatjar in siva-java?

File: project/plugins.sbt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.5")
```
Comment (erizocosmico) [inline project/plugins.sbt]: we don't need assembly for this project

File: build.sbt [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
     name := "siva-java",
     libraryDependencies += scalaTest % Test,
-    libraryDependencies += commonsIO % Test
-  )
+    libraryDependencies += scoverage % Test,
+    libraryDependencies += commonsIO % Test,
 
-jacoco.settings
+    test in assembly := {},
+    assemblyJarName in assembly := s"${name.value}-uber.jar"
```
Comment (erizocosmico) [inline build.sbt]: this is not needed

File: build.sbt [hunk 4]
```diff
@@ -1,19 +1,21 @@
...
     )),
     name := "siva-java",
     libraryDependencies += scalaTest % Test,
-    libraryDependencies += commonsIO % Test
-  )
+    libraryDependencies += scoverage % Test,
+    libraryDependencies += commonsIO % Test,
 
-jacoco.settings
+    test in assembly := {},
```
Comment (erizocosmico) [inline build.sbt]: this is not needed

**Review by erizocosmico — CHANGES_REQUESTED** [review]
This library does not need a fatjar, so basically removing everything that has to do with assembly

**Comment by mcarmonaa** [comment]
Uneeded `fatjar` generation things have been removed

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        SCALA_VERSION := $(TRAVIS_SCALA_VERSION)
+endif
+
+SBT = ./sbt ++$(SCALA_VERSION)
+
+
+# Rules
+all: clean build
+
+.PHONY: fatjar test clean
```
Comment (bzz) [inline Makefile]: nasty `fatjar` sneaked in 🐌

**Review by bzz — APPROVED** [review]
👍 for eclipse support `sbt` convenience wrapper. 
Do you think we could update README with `sbt` -> `./sbt` as well?

LGTM as soon as for few minor things above adressed.

**Review by erizocosmico — APPROVED** [review]

**Comment by erizocosmico** [comment]
LGTM after @bzz's requests have been addressed

**Comment by mcarmonaa** [comment]
@bzz about changes in README, I'll do another PR updating documentation (README and javadoc)

**Review by bzz — APPROVED** [review]
LGTM. 

Let's :shipit: !

**Pull request merged**